source/common/upstream/load_balancer_impl.h

```diff
@@ -485,17 +529,17 @@ class RoundRobinLoadBalancer : public EdfLoadBalancerBase {
  *    The benefit of the Maglev table is at the expense of resolution, memory usage is capped.
  *    Additionally, the Maglev table can be shared amongst all threads.
  */
-class LeastRequestLoadBalancer : public EdfLoadBalancerBase,
-                                 Logger::Loggable<Logger::Id::upstream> {
+class LeastRequestLoadBalancer : public EdfLoadBalancerBase {
```

Comment:
@tonya11en since we are modifying `hostWeight` function in both RR and LR, it would be odd/inconsistent to only subclass LR type. I agree that it would make code more readable, but it also brings code complexity in other part of codebase. In code part where LBs are created we will need to check if slow start config is defined and then create subclass types of RR/LR instead of regular types. I would leave this code as it is, but what do you think?

Reply:
Yeah, that makes sense to me.